src/components/NewsCard/NewsCard.js

```diff
@@ -1,39 +1,89 @@
 import './NewsCard.css';
-import React from 'react';
+import { useEffect, useState } from 'react';
 import { useIsSavedPage } from '../../contexts/IsOnSavedPageContext';
 import { useAuth } from '../../contexts/AuthContext';
+import { localizeDate } from '../../utils/utils';
+import defaultCardImage from '../../images/default_card_image.jpg';
+import { usePopups } from '../../contexts/PopupsContext';
+import { useArticles } from '../../contexts/ArticlesContext';
 
 function NewsCard({ card }) {
   const { isSavedPage } = useIsSavedPage();
   const { isLoggedIn } = useAuth();
+  const [formattedCard, setFormattedCard] = useState({});
+  const { keyword, handleSaveArticle, isArticleSaved, handleDeleteArticle } =
+    useArticles();
+  const { openSignInPopup } = usePopups().popups.signInPopup;
+
+  useEffect(() => {
+    if (!isSavedPage) {
+      setFormattedCard({
+        title: card.title,
+        text: card.description,
+        date: localizeDate(card.publishedAt),
+        source: card.source.name,
+        link: card.url,
+        image: card.urlToImage,
+        keyword: keyword,
+      });
+    } else {
+      setFormattedCard(card);
+    }
+  }, [card, isSavedPage]);
+
+  function onBookmarClick(e) {
+    e.stopPropagation();
+    if (!isLoggedIn) {
+      openSignInPopup();
+      return;
+    }
+    if (isArticleSaved(formattedCard)) {
+      handleDeleteArticle(isArticleSaved(formattedCard));
+    } else {
+      handleSaveArticle(formattedCard);
+    }
+  }
+
+  function onDeleteClick(e) {
+    e.stopPropagation();
+    handleDeleteArticle(card);
+  }
 
   return (
     <li
       className='card'
       onClick={() => {
```

Comment:
The criteria say:


- [] <!--1.88--> Arrow functions are not passed as handler functions.



but you pass here an arrow function, so it's better to define it and then pass the name here